components/eventing-controller/pkg/handlers/jetstream_test.go

```diff
@@ -961,196 +963,140 @@ func TestJSSubscriptionUsingCESDK(t *testing.T) {
+				jsBackend.config.MaxReconnects = tc.maxReconnects
+			}
+			initErr := jsBackend.Initialize(nil)
+			require.NoError(t, initErr)
+
+			// Create a subscription
+			subName := fmt.Sprintf("%s%d", "sub", id)
+			sub := evtesting.NewSubscription(subName, "foo",
+				evtesting.WithNotCleanFilter(),
+				evtesting.WithSinkURL(subscriber.SinkURL),
+				evtesting.WithStatusConfig(defaultSubsConfig),
+			)
+			addJSCleanEventTypesToStatus(sub, testEnvironment.cleaner, jsBackend)
+
+			// when
+			err := jsBackend.SyncSubscription(sub)
+
+			// then
+			require.NoError(t, err)
 
-	// check that the stream is still present
-	_, err = testEnvironment.jsClient.StreamInfo(defaultStreamName)
-	require.NoError(t, err)
+			ev1data := fmt.Sprintf("%s%d", "sampledata", id)
+			require.NoError(t, SendEventToJetStream(jsBackend, ev1data))
+			expectedEv1Data := fmt.Sprintf("\"%s\"", ev1data)
+			require.NoError(t, subscriber.CheckEvent(expectedEv1Data))
+
+			// given
+			testEnvironment.natsServer.Shutdown()
+			require.Eventually(t, func() bool {
+				return !jsBackend.conn.IsConnected()
+			}, 30*time.Second, 2*time.Second)
+
+			// when
+			_ = evtesting.RunNatsServerOnPort(
+				evtesting.WithPort(testEnvironment.natsPort),
+				evtesting.WithJetStreamEnabled())
+
+			// then
+			if tc.maxReconnects == defaultMaxReconnects {
+				require.Eventually(t, func() bool {
+					return jsBackend.conn.IsConnected()
+				}, 30*time.Second, 2*time.Second)
+			}
+
+			_, err = testEnvironment.jsClient.StreamInfo(defaultStreamName)
+			if strings.EqualFold(tc.storageType, JetStreamStorageTypeMemory) &&
+				tc.maxReconnects != defaultMaxReconnects {
+				// for memory storage with reconnects disabled
+				require.True(t, errors.Is(err, nats.ErrStreamNotFound))
+			} else {
+				// check that the stream is still present for file storage
+				// or recreated via reconnect handler for memory storage
+				require.NoError(t, err)
+			}
+
+			if !(strings.EqualFold(tc.storageType, JetStreamStorageTypeFile) &&
+				tc.maxReconnects == defaultMaxReconnects) {
+				// sync the subscription again to recreate invalid subscriptions or consumers
+				err = jsBackend.SyncSubscription(sub)
+				require.NoError(t, err)
+			}
+
+			// stream exists
+			_, err = testEnvironment.jsClient.StreamInfo(defaultStreamName)
+			require.NoError(t, err)
 
-	ev2data := "newsampledata"
-	require.NoError(t, SendEventToJetStream(jsBackend, ev2data))
-	expectedEv2Data := fmt.Sprintf("\"%s\"", ev2data)
-	require.NoError(t, subscriber.CheckEvent(expectedEv2Data))
+			ev2data := fmt.Sprintf("%s%d", "newsampledata", id)
+			require.NoError(t, SendEventToJetStream(jsBackend, ev2data))
+			expectedEv2Data := fmt.Sprintf("\"%s\"", ev2data)
+			require.NoError(t, subscriber.CheckEvent(expectedEv2Data))
 
-	// cleanup the stream from file storage
-	err = testEnvironment.jsClient.DeleteStream(defaultStreamName)
-	require.NoError(t, err)
+			if strings.EqualFold(tc.storageType, JetStreamStorageTypeFile) {
```

Comment:
Yeah I switched to simple equality checks now.

We need to clean up the file storage stream, since they are created in the local temp file and maybe used across other tests, hence leading to inaccurate tests.

 As discussed I put the line to cleanup streams as a defer statement so its done at the end of all tests, irrespective of the stream present in the file or not.